bson/unmarshal_test.go

```diff
@@ -10,64 +10,42 @@ import (
 	"reflect"
 	"testing"
 
-	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/assert"
 	"go.mongodb.org/mongo-driver/bson/bsoncodec"
 	"go.mongodb.org/mongo-driver/bson/bsonrw"
-	"go.mongodb.org/mongo-driver/internal/testutil/assert"
 	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
 )
 
 func TestUnmarshal(t *testing.T) {
-	for _, tc := range unmarshalingTestCases {
+	for _, tc := range unmarshalingTestCases() {
 		t.Run(tc.name, func(t *testing.T) {
-			if tc.reg != nil {
-				t.Skip() // test requires custom registry
-			}
 			got := reflect.New(tc.sType).Interface()
 			err := Unmarshal(tc.data, got)
 			noerr(t, err)
-			if !cmp.Equal(got, tc.want) {
-				t.Errorf("Did not unmarshal as expected. got %v; want %v", got, tc.want)
-			}
+			assert.Equal(t, tc.want, got, "Did not unmarshal as expected.")
 		})
 	}
 }
 
 func TestUnmarshalWithRegistry(t *testing.T) {
-	for _, tc := range unmarshalingTestCases {
+	for _, tc := range unmarshalingTestCases() {
 		t.Run(tc.name, func(t *testing.T) {
-			var reg *bsoncodec.Registry
-			if tc.reg != nil {
-				reg = tc.reg
-			} else {
-				reg = DefaultRegistry
-			}
 			got := reflect.New(tc.sType).Interface()
-			err := UnmarshalWithRegistry(reg, tc.data, got)
+			err := UnmarshalWithRegistry(DefaultRegistry, tc.data, got)
```

Comment:
Since this uses the `DefaultRegistry`, can `TestUnmarshalWithRegistry` be removed in favor of `TestUnmarshal`?

Reply:
Good point. Leaving `TestUnmarshalWithRegistry` seems preferable.